shared/src/event_handling.rs

```diff
@@ -33,23 +33,30 @@ where
 /// Databases: might transform, filter and classify which events are inserted
 /// HashSet: For less persistent (in memory) storing, insert events into a set.
 #[async_trait::async_trait]
-pub trait EventStoring<T> {
+pub trait EventStoring<T, U> {
```

Comment:
I'm not sure I understand why the context here is generic. It seems like it has to be `String` for the `EventHandler` so you can't really implement `EventStoring` for anything other than that.

Reply:
I was going to use an enum PoolType for balancer and then () for GPSettelment. but it doesn't matter now, this approach isn't going to work.